src/ui/public/vis/request_handlers/courier.js

```diff
@@ -0,0 +1,51 @@
+import _ from 'lodash';
+
+const CourierRequestHandlerProvider = function (Private, courier, timefilter) {
+  return {
+    name: 'courier',
+    handler: function (vis, appState, uiState, searchSource) {
```

Comment:
We need some way for visualizations to retrigger the request-handler. Right now, it is only triggered on pressing the play-button of the default-editor.

This is required for example in the tilemap-visualizations. We requery for data on zoom-changes. We trigger this new request manually now (https://github.com/thomasneirynck/kibana/blob/ba6c45abdbe288b87785d0792e4961d4b5742daa/src/core_plugins/tile_map/public/maps_visualization.js#L111). But this could be necessary for any visualization that wants to requery-data, due to interaction with the visualizaiton and not due to changes in the editor.

Could we perhaps add some method to the vis-object, something like `invalidateRequest`, which retriggers a request?